pkg/operator/mon/health.go

```diff
@@ -118,6 +127,14 @@ func (c *Cluster) checkHealth() error {
 		}
 	}
 
+	// handle all mons that haven't been in the Ceph mon map
+	for mon := range monsNotFound {
+		// bring up a new mon to replace the "not found" mon
+		if err := c.failoverMon(mon); err != nil {
```

Comment:
it will be possible that failing over the mons will result in more than the desired number of mons so sometimes we might need to call `removeMon()` instead of `failoverMon()`. I'd suggest you factor out the code above into a method that is also called here. The method would be basically lines 114-124.

Reply:
@travisn Good idea. I moved it to an own function named `failMon()` (see https://github.com/rook/rook/pull/1080/files#diff-bde5ec138117f3d7d9ee4274646be734R129).